logout_state.go

```diff
@@ -7,13 +7,25 @@ func (state logoutState) String() string { return "Logout State" }
 func (s logoutState) IsLoggedOn() bool   { return false }
 
 func (state logoutState) FixMsgIn(session *session, msg Message) (nextState sessionState) {
-	return state
+	var msgType FIXString
+	if err := msg.Header.GetField(tagMsgType, &msgType); err == nil {
```

Comment:
For clarity, might be better to return immediately if err!=nil and avoid the else clause